### DISCUSSION TITLE
Responsiveness

### DISCUSSION THREAD

**Discussion by Vandan1423** [twbs/bootstrap#40937] [Q&A]
See the following webpage made purely using bootstrap, When I am checking the responsiveness in the safari it is not responding but when I can checking it in the chrome the responsiveness is working fine can anyone explain me 

https://github.com/user-attachments/assets/1c9a965f-7171-4a69-8ca3-a2811e3845ba

https://github.com/user-attachments/assets/afec2b2e-6a58-468a-bcc1-495b3d1c9fec

My code is -- 
<!DOCTYPE html>
    <html lang="en">

    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width,initial-scale=1.0">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css">
        <title>Pricing Page</title>
    </head>

    <body>
        <h1 class="text-center">Pricing</h1>
        <p class="text-center">Please check out our most effective payment models</p>
        <div class="row text-center justify-content-evenly">
            <div class="card col-sm-5 col-md-3 mb-3 me-3">
                <div class="card-header">
                    <h4>Free</h4>
                </div>
                <div class="card-body">
                    <h1 class="card-title">$0/mo</h1>
                    <ul class="list-unstyled">
                        <li>10 users included</li>
                        <li>2GB of storage</li>
                        <li>Email support</li>
                    </ul>
                    <button href="#" class="btn btn-outline-primary">Sign up for free</button>
                </div>
            </div>
            <div class="card col-sm-5 col-md-3 mb-3 me-3">
                <div class="card-header">
                    <h4>Pro</h4>
                </div>
                <div class="card-body">
                    <h1 class="card-title">$15/mo</h1>
                    <ul class="list-unstyled">
                        <li>20 users included</li>
                        <li>10GB of storage</li>
                        <li>Priority email support</li>
                    </ul>
                    <button href="#" class="btn btn-primary">Get started</button>
                </div>
            </div>
            <div class="card col-sm-10 col-md-3 mb-3 me-3">
                <div class="card-header">
                    <h4>Enterprise</h4>
                </div>
                <div class="card-body">
                    <h1 class="card-title">$29/mo</h1>
                    <ul class="list-unstyled">
                        <li>30 users included</li>
                        <li>15 GB of storage</li>
                        <li>Phone & email support</li>
                    </ul>
                    <button href="#" class="btn btn-primary">Contactus</button>
                </div>
            </div>
        </div>
    </body>

    </html>

**Comment by julien-deramond** [comment]
Hey. I've created a [CodePen for your example](https://codepen.io/julien-deramond/pen/VwoPvGO). Seems to be responsive on Safari too.
Which version of macOS/Safari are you using?

**Reply by Vandan1423** [reply]
I am using 18.0.1

**Reply by julien-deramond** [reply]
OK, so not an old version. Tested the CodePen with 17.5 and earlier versions, and it works well. Have you the same issue with the CodePen?

**Reply by Vandan1423** [reply]
Actually with codepen its working, but when I copy and pasted your code in my code editor and open the web page I am facing the same issue, this issue is with this file only not with others, Below is the exact problem what I am facing, actually when I am starting with the minimum size possible its responsive well enough, but as I am decreasing the size of my screen then same issue is there.

https://github.com/user-attachments/assets/aa30b429-1df8-4b52-9625-0b96622d3044

**Reply by julien-deramond** [reply]
That's odd. I launched a real server using the same file setup as you and didn't encounter any issues with Safari 17.5. I also tested it on BrowserStack with the exact same version of Safari you're using, and it worked fine there as well. Honestly, I'm not sure what might be causing the problem on your end. Hopefully, someone else in the community can provide some insights.

**Comment by ynng3** [comment]
I have read the entire conversation, and it doesn't seem to be a code issue.

Are you using any Safari browser extensions that might interfere with normal browsing?

Extensions installed on the user's Safari can impact responsive behavior. Ad blockers or privacy-related extensions, in particular, can interfere with CSS or JavaScript loading.

Please test again with all extensions disabled.
To disable extensions in Safari: Go to the "Safari" menu > "Preferences" > "Extensions" tab and turn off all extensions.

**Comment by aniruddhaadak80** [comment]
Hi @Vandan1423 ,

The responsiveness issue in Safari might be related to the CSS or JavaScript. Here's a quick checklist with code suggestions:

1. **Viewport Meta Tag**: Ensure your viewport tag is correctly set. Yours looks good:
   ```html
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   ```

2. **Flexbox and Grid CSS**: Double-check the Bootstrap classes used for layout. Sometimes, Safari has issues with flex properties. Here’s a simplified structure:
   ```html
   <div class="container">
       <div class="row">
           <div class="col-sm-5 col-md-3 mb-3">...</div>
           <div class="col-sm-5 col-md-3 mb-3">...</div>
           <div class="col-sm-10 col-md-3 mb-3">...</div>
       </div>
   </div>
   ```

3. **CSS Reset**: Sometimes a CSS reset helps:
   ```css
   * {
       margin: 0;
       padding: 0;
       box-sizing: border-box;
   }
   ```

4. **Inspect Element**: Use Safari’s Developer Tools (Right-click > Inspect Element) to check for any CSS conflicts.

5. **Update Safari**: Ensure that you're using the latest version of Safari as compatibility can vary.

If issues persist, consider sharing the specific parts of your CSS that might be causing layout problems.

Thanks!
